pkg/controller/daemon/daemoncontroller.go

```diff
@@ -994,32 +990,37 @@ func (dsc *DaemonSetsController) syncDaemonSet(key string) error {
 
 	if ds.DeletionTimestamp != nil || !dsc.expectations.SatisfiedExpectations(dsKey) {
 		// Only update status.
-		return dsc.updateDaemonSetStatus(ds, hash)
+		return dsc.updateDaemonSetStatus(ds, hash, nodeList)
 	}
 
-	err = dsc.manage(ds, hash)
+	err = dsc.manage(ds, hash, nodeList)
 	if err != nil {
 		return err
+	} else {
```

Comment:
Actually I'm not sure if comments from #47243 are correct. Assuming that `dsc.expectations.SatisfiedExpectations(dsKey)` works as expected then if `mange` adds/delete pods then `rollingUpdate` will not be called. 
@janetkuo can you comment?